### PR TITLE
Bluetooth: controller: HCI stubs for BIG commands

### PR DESCRIPTION
**Pull request by Thalley**
Adds initial HCI stubs for the BIG commands to be used for
broadcast isochronous channels.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

### CODE REVIEW

**Review by kruithofa — APPROVED** [review]
LGTM

File: include/bluetooth/hci.h [hunk 1]
```diff
@@ -2053,6 +2071,23 @@ struct bt_hci_evt_le_req_peer_sca_complete {
 	uint8_t  sca;
 } __packed;
 
+#define BT_HCI_EVT_LE_BIGINFO_ADV_REPORT        0x22
+struct bt_hci_evt_le_biginfo_advertising_report {
```
Comment (wopu-ot) [inline include/bluetooth/hci.h:2075]: Maybe consider shortening "advertising" to "adv"

**Review by wopu-ot — APPROVED** [review]

File: include/bluetooth/hci.h (hunk 1 above)
Comment (Thalley) [reply include/bluetooth/hci.h]: Agreed. Done

File: include/bluetooth/hci.h [hunk 2]
```diff
@@ -2053,6 +2071,23 @@ struct bt_hci_evt_le_req_peer_sca_complete {
...
+	uint16_t iso_interval;
+	uint8_t  bn;
+	uint8_t  pto;
+	uint8_t  irc;
+	uint16_t max_pdu;
+	uint8_t  sdu_interval[3];
+	uint16_t max_sdu;
+	uint8_t  phy;
+	uint8_t  framing;
+	uint8_t  enc;
```
Comment (cvinayak) [inline include/bluetooth/hci.h:2088]:
```suggestion
	uint8_t  encryption;
```

File: include/bluetooth/hci.h [hunk 3]
```diff
@@ -2053,6 +2071,23 @@ struct bt_hci_evt_le_req_peer_sca_complete {
 	uint8_t  sca;
 } __packed;
 
+#define BT_HCI_EVT_LE_BIGINFO_ADV_REPORT        0x22
+struct bt_hci_evt_le_biginfo_adv_report {
+	uint16_t handle;
```
Comment (cvinayak) [inline include/bluetooth/hci.h:2076]:
```suggestion
	uint16_t sync_handle;
```

File: subsys/bluetooth/controller/hci/hci.c [hunk 4]
```diff
@@ -1062,6 +1062,60 @@ static void le_set_adv_enable(struct net_buf *buf, struct net_buf **evt)
...
+	uint8_t status;
+	uint32_t sdu_interval;
+	uint16_t max_sdu;
+	uint16_t max_latency;
+
+	sdu_interval = sys_get_le24(cmd->sdu_interval);
+	max_sdu = sys_le16_to_cpu(cmd->max_sdu);
+	max_latency = sys_le16_to_cpu(cmd->max_latency);
+
+	status = ll_create_big(cmd->big_handle, cmd->adv_handle, cmd->num_bis,
```
Comment (cvinayak) [inline subsys/bluetooth/controller/hci/hci.c:1078]:
```suggestion
	status = ll_big_create(cmd->big_handle, cmd->adv_handle, cmd->num_bis,
```

File: subsys/bluetooth/controller/hci/hci.c [hunk 5]
```diff
@@ -1062,6 +1062,60 @@ static void le_set_adv_enable(struct net_buf *buf, struct net_buf **evt)
...
+	uint16_t iso_interval;
+	uint16_t max_sdu;
+	uint16_t max_pdu;
+
+	sdu_interval = sys_get_le24(cmd->sdu_interval);
+	iso_interval = sys_le16_to_cpu(cmd->iso_interval);
+	max_sdu = sys_le16_to_cpu(cmd->max_sdu);
+	max_pdu = sys_le16_to_cpu(cmd->max_pdu);
+
+	status = ll_create_big_test(cmd->big_handle, cmd->adv_handle,
```
Comment (cvinayak) [inline subsys/bluetooth/controller/hci/hci.c:1100]:
```suggestion
	status = ll_big_test_create(cmd->big_handle, cmd->adv_handle,
```

File: subsys/bluetooth/controller/hci/hci.c [hunk 6]
```diff
@@ -1062,6 +1062,60 @@ static void le_set_adv_enable(struct net_buf *buf, struct net_buf **evt)
...
+
+	*evt = cmd_status(status);
+}
+
+static void le_terminate_big(struct net_buf *buf, struct net_buf **evt)
+{
+	struct bt_hci_cp_le_terminate_big *cmd = (void *)buf->data;
+	uint8_t status;
+
+	status = ll_terminate_big(cmd->big_handle, cmd->reason);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/hci/hci.c:1115]:
```suggestion
	status = ll_big_terminate(cmd->big_handle, cmd->reason);
```

File: subsys/bluetooth/controller/include/ll.h [hunk 7]
```diff
@@ -94,9 +94,27 @@ uint8_t ll_adv_enable(uint8_t handle, uint8_t enable,
...
+		      uint16_t max_latency, uint8_t rtn, uint8_t phy,
+		      uint8_t packing, uint8_t framing, uint8_t encryption,
+		      uint8_t *bcode);
+uint8_t ll_create_big_test(uint8_t big_handle, uint8_t adv_handle,
+			   uint8_t num_bis, uint32_t sdu_interval,
+			   uint16_t iso_interval, uint8_t nse, uint16_t max_sdu,
+			   uint16_t max_pdu, uint8_t phy, uint8_t packing,
+			   uint8_t framing, uint8_t bn, uint8_t irc,
+			   uint8_t pto, uint8_t encryption, uint8_t *bcode);
+uint8_t ll_terminate_big(uint8_t big_handle, uint8_t reason);
```
Comment (cvinayak) [inline subsys/bluetooth/controller/include/ll.h:108]:
```suggestion
uint8_t ll_big_create(uint8_t big_handle, uint8_t adv_handle, uint8_t num_bis,
		      uint32_t sdu_interval, uint16_t max_sdu,
		      uint16_t max_latency, uint8_t rtn, uint8_t phy,
		      uint8_t packing, uint8_t framing, uint8_t encryption,
		      uint8_t *bcode);
uint8_t ll_big_test_create(uint8_t big_handle, uint8_t adv_handle,
			   uint8_t num_bis, uint32_t sdu_interval,
			   uint16_t iso_interval, uint8_t nse, uint16_t max_sdu,
			   uint16_t max_pdu, uint8_t phy, uint8_t packing,
			   uint8_t framing, uint8_t bn, uint8_t irc,
			   uint8_t pto, uint8_t encryption, uint8_t *bcode);
uint8_t ll_big_terminate(uint8_t big_handle, uint8_t reason);
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Comment by cvinayak** [comment]
@Thalley needs rebase.

File: subsys/bluetooth/controller/CMakeLists.txt [hunk 8]
```diff
@@ -78,6 +78,9 @@ if(CONFIG_BT_LL_SW_SPLIT)
     CONFIG_BT_HCI_MESH_EXT
     ll_sw/ll_mesh.c
     )
+  zephyr_library_sources(
+    ll_sw/ull_iso_broadcast.c
+    )
```
Comment (cvinayak) [inline subsys/bluetooth/controller/CMakeLists.txt:83]: Move into with broadcaster

```suggestion
  zephyr_library_sources_ifdef(
  CONFIG_BT_CTLR_ADV_ISO
    ll_sw/ull_adv_iso.c
    )
```

Move into with observer

```suggestion
  zephyr_library_sources_ifdef(
  CONFIG_BT_CTLR_SYNC_ISO
    ll_sw/ull_sync_iso.c
    )
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/Kconfig [hunk 9]
```diff
@@ -428,6 +436,28 @@ config BT_CTLR_SYNC_PERIODIC
 config BT_CTLR_SYNC_PERIODIC
 	bool "LE Periodic Advertising in Synchronization State [EXPERIMENTAL]" if BT_LL_SW_SPLIT
 
+config BT_CTLR_ADV_ISO
+	bool "LE Broadcast Isochronous Channel advertising [EXPERIMENTAL]" if !BT_LL_SW_SPLIT
```
Comment (cvinayak) [inline subsys/bluetooth/controller/Kconfig:440]:
```suggestion
	bool "LE Broadcast Isochronous Channel advertising" if !BT_LL_SW_SPLIT
```

File: subsys/bluetooth/controller/Kconfig [hunk 10]
```diff
@@ -428,6 +436,28 @@ config BT_CTLR_SYNC_PERIODIC
...
+	select BT_CTLR_ADV_PERIODIC
+	help
+	  Enable support for Bluetooth 5.2 LE Isochronous Advertising in the
+	  Controller.
+
+config BT_CTLR_ADV_ISO
+	bool "LE Broadcast Isochronous Channel advertising [EXPERIMENTAL]" if BT_LL_SW_SPLIT
+
+config BT_CTLR_SYNC_ISO
+	bool "LE Broadcast Isochronous Channel advertising sync [EXPERIMENTAL]" if !BT_LL_SW_SPLIT
```
Comment (cvinayak) [inline subsys/bluetooth/controller/Kconfig:451]:
```suggestion
	bool "LE Broadcast Isochronous Channel advertising sync" if !BT_LL_SW_SPLIT
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Review by cvinayak — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**